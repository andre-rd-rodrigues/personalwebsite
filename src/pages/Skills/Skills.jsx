import soft from "assets/images/soft.jpeg";
import work from "assets/images/work.jpeg";
import AnimatedHeading from "components/AnimatedHeading/AnimatedHeading";
import PageContainer from "components/PageContainer/PageContainer";
import { Col, Row } from "react-bootstrap";
import { useTranslation } from "react-i18next";
import styles from "./skills.module.scss";

const Skills = () => {
  const { t } = useTranslation();

  return (
    <PageContainer className={styles.container}>
      <AnimatedHeading>
        <h1>Skills</h1>
      </AnimatedHeading>
      <Row className={styles.section}>
        <Col sm={12} md={12} lg={6} className={styles.imgCol}>
          <img src={work} alt="Development experience" className={styles.img} />
        </Col>
        <Col sm={12} md={12} lg={6} className={styles.developerSkillsCol}>
          <div>
            <h2>Development</h2>
            <ul>
              <li>
                Master the Basics - <b>HTML5, CSS3, ES6, LESS, SASS, jQuery</b>
              </li>
              <li>
                Deep knowledge in Responsive Layouts -
                <b>Bootstrap, React-Bootstrap, Tailwind-CSS, Media queries</b>
              </li>
              <li>
                Front-End Libraries - <b>React.js, React Native</b>
              </li>
              <li>
                Back-end -
                <b>Node.js, RESTful APIs, GraphQL, MongoDB, Mongoose</b>
              </li>
              <li>
                Infrastructure - <b>Pipelines, yml, Github actions, Docker</b>
              </li>
              <li>
                Testing - <b>Jest, Enzyme, React Testing Library</b>
              </li>
              <li>
                Others -
                <b>
                  Next, Version control (Git/Github), Python, Webpack and others
                </b>
              </li>
            </ul>
          </div>
          <div className={styles.webDesignContainer}>
            <h2>Web design</h2>
            <ul>
              <li>Sketch</li>
              <li>Figma</li>
              <li>Affinity Designer</li>
              <li>Affinity Publisher</li>
              <li>Adobe Illustrator</li>
              <li>UX, UI, Web, Mobile, Apps, Logos</li>
            </ul>
          </div>
        </Col>
      </Row>

      <Row>
        <Col sm={12} md={12} lg={6} className={styles.softSkillsCol}>
          <div>
            <h2>Soft skills</h2>
            <p>{t("skills_page.soft_skills")}</p>
          </div>
        </Col>
        <Col sm={12} md={12} lg={6} className={styles.softSkillsImgCol}>
          <img src={soft} alt="Development experience" className={styles.img} />
        </Col>
      </Row>
    </PageContainer>
  );
};

export default Skills;
