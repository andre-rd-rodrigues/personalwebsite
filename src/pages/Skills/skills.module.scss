@import "assets/styles/mixins";

.container {
  li {
    font-weight: 300;
    margin-bottom: 15px;
  }
  h1 {
    font-size: var(--fs-xxl);
    margin-bottom: 40px;
  }
}

@include title_transparent;

.section {
  margin-bottom: 90px;
  h2 {
    margin-bottom: 20px;
  }
}

.developerSkillsCol {
  padding-left: 40px !important;
}

.webDesignContainer {
  h2 {
    margin-top: 30px;
  }
}

.img {
  width: 100%;
  max-height: 700px;
  object-fit: cover;
  border-radius: 10px;
}

.downloadCVContainer {
  text-align: center;
  margin-top: 70px;
}

@media only screen and (max-width: $breaking-point-lg) {
  .developerSkillsCol {
    padding-left: 12px !important;
  }
  .imgCol {
    margin-bottom: 40px;
  }

  .softSkillsCol {
    order: 2;
  }

  .softSkillsImgCol {
    margin-bottom: 40px;
    order: 1;
  }

  .webDesignContainer {
    h2 {
      margin-top: 40px;
    }
  }

  .section {
    margin-bottom: 40px;
  }
}
