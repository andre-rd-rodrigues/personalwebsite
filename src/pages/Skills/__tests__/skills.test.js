import { render } from "utils/tests/test-utils";
import Skills from "../Skills";

describe("Skills page", () => {
  it("renders page correctly", () => {
    const { asFragment } = render(<Skills />);

    expect(asFragment()).toMatchSnapshot();
  });
});
