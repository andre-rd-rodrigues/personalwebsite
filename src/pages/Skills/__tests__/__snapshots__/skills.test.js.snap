// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills page renders page correctly 1`] = `
<DocumentFragment>
  <div
    class="container container"
  >
    <div
      class="wrapper"
      style="opacity: 0;"
    >
      <div
        class="container undefined"
      >
        <span
          style="white-space: nowrap;"
        >
          <span
            style="overflow: hidden; display: inline-block;"
          >
            <h1
              style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
            >
              S
            </h1>
          </span>
          <span
            style="overflow: hidden; display: inline-block;"
          >
            <h1
              style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
            >
              k
            </h1>
          </span>
          <span
            style="overflow: hidden; display: inline-block;"
          >
            <h1
              style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
            >
              i
            </h1>
          </span>
          <span
            style="overflow: hidden; display: inline-block;"
          >
            <h1
              style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
            >
              l
            </h1>
          </span>
          <span
            style="overflow: hidden; display: inline-block;"
          >
            <h1
              style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
            >
              l
            </h1>
          </span>
          <span
            style="overflow: hidden; display: inline-block;"
          >
            <h1
              style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
            >
              s
            </h1>
          </span>
        </span>
      </div>
      <div
        class="section row"
      >
        <div
          class="imgCol col-lg-6 col-md-12 col-sm-12"
        >
          <img
            alt="Development experience"
            class="img"
            src="/images/skills/development.jpg"
            style="opacity: 0;"
          />
        </div>
        <div
          class="developerSkillsCol col-lg-6 col-md-12 col-sm-12"
        >
          <div
            style="opacity: 0;"
          >
            <h2>
              Development
            </h2>
            <ul>
              <li>
                <b>
                  Master the Basics:
                </b>
                 HTML5, CSS3, Javascript, LESS, SASS, jQuery
              </li>
              <li>
                <b>
                  Deep knowledge in Responsive Layouts:
                </b>
                 Bootstrap, React-Bootstrap, Tailwind-CSS, Media queries
              </li>
              <li>
                <b>
                  Front-End Frameworks:
                </b>
                 React.js, React Native, Next.js
              </li>
              <li>
                <b>
                  Back-end:
                </b>
                 Node.js, RESTful APIs, GraphQL, MongoDB, Mongoose
              </li>
              <li>
                <b>
                  DevOps:
                </b>
                 Github actions, Docker, Pipelines, yml
              </li>
              <li>
                <b>
                  Testing:
                </b>
                 Jest, Enzyme, React Testing Library
              </li>
              <li>
                <b>
                  Others:
                </b>
                 Version control (Git/Github), Python, Webpack and others
              </li>
            </ul>
          </div>
          <div
            class="webDesignContainer"
            style="opacity: 0;"
          >
            <h2>
              Web design
            </h2>
            <ul>
              <li>
                Sketch
              </li>
              <li>
                Figma
              </li>
              <li>
                Affinity Designer
              </li>
              <li>
                Affinity Publisher
              </li>
              <li>
                Adobe Illustrator
              </li>
              <li>
                UX, UI, Web, Mobile, Apps, Logos
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="softSkillsCol col-lg-6 col-md-12 col-sm-12"
        >
          <div
            style="opacity: 0;"
          >
            <h2>
              Soft skills
            </h2>
            <p>
              Arts and human-behavior had always been a huge part of my life. Through my other passions like theatre, music, and psychology, I was able to develop versatile skills that contributed immensely to my emotional, social, and psychological health, revealing an asset in the work environment. Some of them are creativity, emotional intelligence, empathy, interpersonal skills, critical thinking, problem-solving, and communication.
            </p>
          </div>
        </div>
        <div
          class="softSkillsImgCol col-lg-6 col-md-12 col-sm-12"
        >
          <img
            alt="Development experience"
            class="img"
            src="/images/skills/soft.jpeg"
            style="opacity: 0;"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
